src/middleware/rateLimiter.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response, NextFunction } from 'express';
+import Redis from 'ioredis';
+import { RateLimiterRedis } from 'rate-limiter-flexible';
+
+import config from 'config';
+import { logger } from 'utils/logger';
+
+interface RateLimiterOptions {
+  keyPrefix: string;
+  points: number;
+  duration: number;
+}
+
+const redisClient = new Redis({
+  host: config.redisHost || 'node-api-redis',
+  port: config.redisPort || 6379,
+  db: config.redisDb || 0,
+  enableOfflineQueue: false,
+});
```

Comment:
Reuse existing database config

```suggestion
const redisClient = new Redis(config.redisCacheUrl);
```